relays/ethereum/src/rpc_errors.rs

```diff
@@ -58,12 +55,28 @@ impl From<SubstrateNodeError> for RpcError {
 	}
 }
 
-impl From<RpcHttpError> for RpcError {
-	fn from(err: RpcHttpError) -> Self {
+impl From<RequestError> for RpcError {
+	fn from(err: RequestError) -> Self {
 		Self::Request(err)
 	}
 }
 
+impl From<ethabi::Error> for RpcError {
+	fn from(err: ethabi::Error) -> Self {
+		Self::Ethereum(EthereumNodeError::ResponseParseFailed(format!("{}", err)))
+	}
+}
+
+impl MaybeConnectionError for RpcError {
+	fn is_connection_error(&self) -> bool {
+		match *self {
+			// Q: Is blanket handling `Request` too broad?
```

Comment:
@svyatonik wdyt?

Reply:
I'm not sure, but imo we should only return true if we see `Request(RequestError::TransportError(_))` here. The idea is that if there are some problems with **connection**, we sleep some time and then retry.